src/GoogleAuthManager.php

```diff
@@ -3,93 +3,188 @@
+   *
+   * @return $this
+   *   The current object.
    */
   public function authenticate() {
-    $this->client->setAccessToken($this->getAccessToken());
-    return $this;
+    $this->token = $this->client->getAccessToken('authorization_code',
+      ['code' => $_GET['code']]);
+
+    return $this->token;
   }
 
   /**
-   * {@inheritdoc}
+   * Gets the data by using the access token returned.
+   *
+   * @return array
+   *   User info returned by the Google.
    */
-  public function getAccessToken() {
-    if (!$this->accessToken) {
-      $this->accessToken = $this->client->fetchAccessTokenWithAuthCode($this->getCode());
-    }
+  public function getUserInfo() {
+    $this->user = $this->client->getResourceOwner($this->token);
+    return $this->user;
+  }
 
-    return $this->accessToken;
+  /**
+   * Gets the data by using the access token returned.
+   *
+   * @return string
+   *   Data returned by Making API Call.
+   */
+  public function getExtraDetails($url) {
+    $httpRequest = $this->client->getAuthenticatedRequest('GET', $url, $this->token, []);
+    $data = $this->client->getResponse($httpRequest);
+    return json_decode($data->getBody(), true);
   }
 
   /**
-   * {@inheritdoc}
+   * Returns the Google login URL where user will be redirected.
+   *
+   * @return string
+   *   Absolute Google login URL where user will be redirected
    */
-  public function getUserInfo() {
-    return $this->getOauth2Service()->userinfo->get();
+  public function getGoogleLoginUrl() {
+    $scopes = [
+      'email',
+      'openid',
+      'profile',
+    ];
+
+    $google_scopes = explode(',', $this->getScopes());
+    foreach ($google_scopes as $scope) {
+      array_push($scopes, $scope);
+    }
+
+    $login_url = $this->client->getAuthorizationUrl([
+      'scope' => $scopes,
+    ]);
+
+    // Generate and return the URL where we should redirect the user.
+    return $login_url;
   }
 
   /**
-   * Gets Google Oauth2 Service.
+   * Returns the Google login URL where user will be redirected.
    *
-   * @return Google_Service_Oauth2
-   *   The Google Oauth2 service.
+   * @return string
+   *   Absolute Google login URL where user will be redirected
    */
-  protected function getOauth2Service() {
-    return new Google_Service_Oauth2($this->getClient());
+  public function getState() {
+    $state = $this->client->getState();
+
+    // Generate and return the URL where we should redirect the user.
+    return $state;
   }
 
   /**
-   * Gets the code returned by Google to authenticate.
+   * Gets the data Point defined the settings form page.
    *
    * @return string
-   *   The code string returned by Google.
+   *   Data points separtated by comma.
    */
-  protected function getCode() {
-    if (!$this->code) {
-      $this->code = $this->request->query->get('code');
+  public function getScopes() {
+    if (!$this->scopes) {
+      $this->scopes = $this->setting->get('scopes');
```

Comment:
Property setting was not defined in the class, please do it.

Btw, I think config sounds better that setting. I think of the Drupal site settings when I see this property, and this object isn't about those settings